### PR TITLE
[FIX] do not generate payment if amount=0

### PR DESCRIPTION
**Pull request by damdam-s**

### CODE REVIEW

**Comment by gurneyalex** [comment]
:+1:

**Comment by pedrobaeza** [comment]
It's a WIP? Anyway, it seems :+1:

**Comment by lepistone** [comment]
:+1: in principle, thanks. Still I am unsure: should we consider a near-zero amount because of float or rounding problems?

**Comment by lepistone** [comment]
closed by mistake, I reopen

**Comment by mdietrichc2c** [comment]
I agree with @lepistone : I think we should consider using `float_is_zero`, especially since the test was added to only call the method when the amount is not 0.

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by mdietrichc2c** [comment]
Actually, :-1: : I didn't notice that the condition was `if float_is_zero`, meaning the call is only made when the amount is equal to 0.
@damdam-s : can you add the `not` in front of `float_is_zero` ?

**Comment by damdam-s** [comment]
@mdietrichc2c ouch ... :( sorry. fix done

File: sale_payment_method/sale.py [hunk 1]
```diff
@@ -103,8 +104,10 @@ def automatic_payment(self, amount=None):
             amounts = [(date, amount)]
 
         # reversed is cosmetic, compute returns terms in the 'wrong' order
+        precision = self.env['decimal.precision'].precision_get('Account')
```
Comment (lepistone) [inline sale_payment_method/sale.py:106]: please move this up 1 line, otherwise the comment does not make sense.

**Comment by lepistone** [comment]
A minor remark, otherwise :+1: thanks!

**Comment by damdam-s** [comment]
@lepistone done

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
:+1:

**Pull request merged**